integration/spark/src/test/resources/spark_scripts/spark_cached.py

```diff
@@ -0,0 +1,25 @@
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import time
+
+os.makedirs("/tmp/caching", exist_ok=True)
+
+from pyspark.sql import SparkSession
+
+spark = SparkSession.builder \
```

Comment:
Can we also add the test cases described in https://github.com/OpenLineage/OpenLineage/issues/610#issuecomment-1133479889 ?

Reply:
I included the test within `SparkReadWriteIntegTest` class and it is passing although I needed to change amount of events emmitted from `18` to `atLeast(12)`. The crucial part, that verifies inputs & outputs, is working properly. 

I also changed test name as I it does not seem to make use of `sqlite` nor `jdbc`.